conn.go

```diff
@@ -111,13 +114,22 @@ func createConn(nextConn net.Conn, flightHandler flightHandler, handshakeMessage
 		loggerFactory = logging.NewDefaultLoggerFactory()
 	}
 
+	connectTimeout := defaultConnectTimeout
+	if config.ConnectTimeout != nil {
+		connectTimeout = *config.ConnectTimeout
+		if connectTimeout <= 0 {
+			connectTimeout = math.MaxInt64 * time.Nanosecond
+		}
```

Comment:
I would move the nested `if` out of it and one up, so you end up with:

```
connectTimeout := defaultConnectTimeout
if config.ConnectTimeout != nil {
	connectTimeout = *config.ConnectTimeout
}
if connectTimeout <= 0 {
    connectTimeout = math.MaxInt64 * time.Nanosecond
}
```

This keeps the flow a bit simpler while achieving the same thing.

Reply:
Yep, that does feel cleaner, thanks! Fixed up